fractal_server/app/routes/api/v2/workflow.py

```diff
@@ -319,8 +319,43 @@ async def import_workflow(
     # Insert tasks
 
     for wf_task in workflow.task_list:
-        source = wf_task.task.source
-        task_id = source_to_id[source]
+        if isinstance(wf_task.task, TaskImportV2Legacy):
+            stm = select(TaskV2).where(TaskV2.source == wf_task.task.source)
+        else:
+            stm = (
+                select(TaskV2)
+                .join(TaskGroupV2)
+                .where(TaskGroupV2.pkg_name == wf_task.task.pkg_name)
+                .where(TaskV2.name == wf_task.task.name)
+            )
+            if wf_task.task.version is not None:
+                stm = stm.where(TaskGroupV2.version == wf_task.task.version)
+            else:
+                stm = stm.order_by(TaskGroupV2.timestamp_created.desc()).limit(
```

Comment:
When entering the "new" branch

First, check if version is set or not in the payload.

IF VERSION NOT SET

1. Find all matching tasks (with no special order) with the current `stm`.
2. Find all their versions.
3. Find latest version (string comparison) or None if no task has a version set.
4. Set version to this specific value.

NOW VERSION IS SET, PROCEED

1. Prepare full query (including version).
2. If results are more than one, disambiguate by user/group ownership